tuner/tuner/merge_td_specs.py

```diff
@@ -0,0 +1,145 @@
+        return top_module
+
+
+def merge_tuning_specs(tuner_ctx: TunerContext, td_specs: list[ir.Module]) -> ir.Module:
+    """
+    Merges multiple input modules (`td_specs`) into a single tuning specification module.
+    First, the input modules are combined into a container module. Then, the external
+    `iree-opt` tool is invoked with the `--iree-codegen-link-tuning-specs` pass to
+    link or merge the individual tuning specs. When all input specs are marked with the
+    default attribute `iree_codegen.tuning_spec_with_default_entrypoint`, they are merged
+    into one tuning spec.
+    """
+    module = combine_tuning_specs(tuner_ctx, td_specs)
+    iree_opt = ireec.binaries.find_tool("iree-opt")
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        input_path = os.path.join(tmpdir, "tmp_input.mlir")
+        output_path = os.path.join(tmpdir, "tmp_output.mlir")
+
+        with open(input_path, "w") as f:
+            f.write(str(module))
+
+        result = subprocess.run(
+            [
+                iree_opt,
+                "--iree-codegen-link-tuning-specs",
+                input_path,
+                "-o",
+                output_path,
+            ],
+            capture_output=True,
+            text=True,
+        )
+
+        if result.returncode != 0:
+            raise RuntimeError(f"iree-opt failed: {result.stderr}")
+
+        with open(output_path, "r") as f:
+            output_mlir = f.read()
+            return ir.Module.parse(output_mlir, tuner_ctx.mlir_ctx)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        prog="merge_td_specs",
+        description="""
+        Merge multiple tuning specification modules into a single one.
+
+        This script wraps the `--iree-codegen-link-tuning-specs` pass and is useful for merging
+        multiple tuning specs into one for further compilation or tuning.
+
+        Examples:
+            python -m tuner.merge_td_specs td_spec_1.mlir td_spec_2.mlir -o merged.mlir
+        """,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "inputs", nargs="+", help="Input MLIR tuning spec files to merge"
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="Output path for merged MLIR file (if omitted, prints to stdout)",
+    )
+
+    parser.add_argument(
+        "--verbose", "-v", action="store_true", help="Enable verbose output to stdout"
+    )
+
+    args = parser.parse_args()
+    tune_logger.setLevel(logging.DEBUG if args.verbose else logging.INFO)
+
+    formatter = logging.Formatter("%(message)s")
+
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatter)
+    tune_logger.addHandler(console_handler)
+
+    with TunerContext() as tuner_ctx:
+        td_specs = []
+        for input_path in args.inputs:
+            tune_logger.info(f"Reading td spec: {input_path}")
```

Comment:
Is this printed by default? If one of your supported output options is to print to stdout, you we can't interleave that with random logs